core/src/apps/bitcoin/sign_tx/zcash.py

```diff
@@ -34,32 +34,23 @@
 OVERWINTERED = const(0x80000000)
 
 
-class Overwintered(Bitcoinlike):
+class Zcashlike(Bitcoinlike):
     def __init__(self, tx: SignTx, keychain: Keychain, coin: CoinInfo) -> None:
         ensure(coin.overwintered)
         super().__init__(tx, keychain, coin)
 
-        if self.tx.version == 3:
-            if not self.tx.branch_id:
-                self.tx.branch_id = 0x5BA81B19  # Overwinter
-        elif self.tx.version == 4:
-            if not self.tx.branch_id:
-                self.tx.branch_id = 0x76B809BB  # Sapling
-        else:
+        if self.tx.version != 4 or self.tx.branch_id is None:
             raise wire.DataError("Unsupported version for overwintered transaction")
 
     async def step8_finish(self) -> None:
+        ensure(self.tx.version == 4)
```

Comment:
i don't think this is necessary, we reject versions != 4 in `__init__`.
i think the original check was mostly just so that there's an `else` case to the options; with no options, we don't need to keep it

Reply:
Fixed 016af57.